Algorithm.CSharp/OptionStrategyFilteringUniverseBoxSpreadRegressionAlgorithm.cs

```diff
@@ -0,0 +1,123 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using QuantConnect.Data.Market;
+using QuantConnect.Securities.Option;
+
+namespace QuantConnect.Algorithm.CSharp
+{
+    /// <summary>
+    /// Regression algorithm of filtering with Box Spread option strategy and asserting it's being detected by Lean and works as expected
+    /// </summary>
+    public class OptionStrategyFilteringUniverseBoxSpreadRegressionAlgorithm : OptionStrategyFilteringUniverseBaseAlgorithm
+    {
+        public override void Initialize()
+        {
+            _func = u => u.IncludeWeeklys().Strikes(-10, +10).Expiration(0, 180).BoxSpread(30, 5);
```

Comment:
Isn't `Expiration` filter overlapping with the `daysTillExpiry` in the option strategy? For example in this example if Expiration was (30, 180) 🤔
Makes me think that `daysTillExpiry` would only be used if `Expiration` filter isn't used? which could mean we can push it to the end of the option strategy as an optional probably defaulting to null, maybe if `Expiration` wasn't use use 30 instead as default

Reply:
It is intentional: `Expiration` is a strict rule, but `daysTillExpiry` is looking for the best match only🤔 I don't think they are mutaully exclusive.

Reply:
> Expiration is a strict rule, but daysTillExpiry is looking for the best match only🤔 

Sounds like a bug if user provides 30 `daysTIllExpiry` and it can use something bellow that?

Reply:
> .Strikes(-10, +10).Expiration(0, 180)

Let's remove this from the examples

~Sounds like a bug if user provides 30 daysTIllExpiry and it can use something bellow that?~ nevermind 👍